### DISCUSSION TITLE
How to import `Duration` used by Rocket?

### DISCUSSION THREAD

**Discussion by H36615** [rwf2/Rocket#2048] [Questions] [answered]
Hi,

Not sure how I should be able to use `time` crate with Rocket.

Just trying to simply set max_age to my cookie as shown in this example https://docs.rs/cookie/0.14.4/cookie/struct.CookieBuilder.html#example. However, I apparently cannot import `Duration` from `rocket`.

If I use separately imported `time` crate in my `Cargo.toml` I get `mismatched types` error. I think there's small chance this is related to https://github.com/SergioBenitez/Rocket/discussions/2026.

How can I make the linked example work on my end?

**Comment by H36615** [comment]
I was able to make it work by downgrading `time` in my `Cargo.toml` from 0.3 -> 0.2 like the Rocket has. So am I supposed to have the `time` dependency in my dependencies at all and manually match the dependency versions (e.g. with Rocket)?

**Reply by somehowchris** [reply]
If you are on `0.14.4` Yes, if you are on a newer version time is reexported

**Reply by H36615** [reply]
I'm on `0.5.0-rc.1`. Sorry, what does it mean that it's re-exported? Can I import `time` when I have just `rocket` in my dependencies?

**Reply by somehowchris** [reply]
Where do you see rocket using `Duration` Or the need for it?

**Reply by H36615** [reply]
In the link I posted in my OP. `.max_age(Duration::days(1))`

**Reply by somehowchris** [reply]
Well that’s not a rocket example

**Reply by somehowchris** [reply]
Anyway rocket 0.5-rc.1 uses `cookie@0.15.1` Which uses `time@0.2.1`, so no reexport of time

**Reply by H36615** [reply]
Oh damn :/ yeah you're right. Don't even have good excuse for that. The explanation was helpful though.

**Comment by H36615** [comment, marked as answer]
As stated by @somehowchris, `time` is not re-exported in this case.
